lib/bank_account/checking_account.rb

```diff
@@ -0,0 +1,44 @@
+module BankAccount
+  class CheckingAccount < Account
+  attr_accessor :check_count, :balance, :id
+
+    def initialize(id, initial_balance)
+      if initial_balance < 0
+        raise ArgumentError.new("You cannot open an account with a negative balance.")
+      else
+      super
```

Comment:
This is a nice use of the base classes functionality - but you could remove lines 6-8 because the account classes initialize method already has that functionality.